floodlight/core/pitch.py

```diff
@@ -147,3 +153,967 @@ def center(self):
+            must be True. If not given as an argument the pitch is not going to be
+            saved. Plots are going to be stored as .png on the same directory layer as
+            the repository.
+        kwargs:
+            You may pass optional arguments (`linewidth`} which can be used for the plot
+            with matplotlib.
+        ax: plt.axes
+            Axes from matplotlib library on which the handball field is plotted.
+
+        Returns
+        ----------
+        ax : matplotlib.axes._subplots.AxesSubplot
+            An axes to which all elements of the sport specific pitch are added.
+        """
+
+        # customizing visualization
+        if color_scheme == "black_white":
+            ax.set_facecolor("white")
+            outline = "black"
+            innerline = "black"
+            fill = "white"
+            goal = "black"
+        else:
+            ax.set_facecolor("skyblue")
+            outline = "white"
+            innerline = "white"
+            fill = "khaki"
+            goal = "white"
+
+        # key data for the handball pitch
+        xmin, xmax = self.xlim
+        ymin, ymax = self.ylim
+
+        x_range = xmax - xmin
+        y_range = ymax - ymin
+        x_half = (xmax + xmin) / 2
+        y_half = (ymax + ymin) / 2
+
+        linewidth = kwargs.get("linewidth", 1)
+
+        # points on the field which are used for plotting all lines and arcs
+        left_side_right_goal_post = (xmin, ymin + y_range * 0.425)
+        left_side_left_goal_post = (xmin, ymin + y_range * 0.575)
+        right_side_left_goal_post = (xmax, ymin + y_range * 0.425)
+        right_side_right_goal_post = (xmax, ymin + y_range * 0.575)
+        width_hight_free_throw_line = y_range * 0.9
+        width_hight_goal_area_line = y_range * 0.6
+        radius_goal_area_line_arc = y_range * 0.3
+        goal_width = y_range * 0.15
+        goal_height = y_range * 0.1
+        y_center_to_post = goal_width / 2
+        y_value_goal_lower_post = y_half - y_center_to_post
+        y_value_goal_upper_post = y_half + y_center_to_post
+        y_pos_lower_goal_posts = ymin + y_range * 0.425
+        y_pos_upper_goal_posts = ymin + y_range * 0.575
+        left_side_x_pos_goal_area_line_edge = xmin + (x_range * 0.15)
+        right_side_x_pos_goal_area_line_edge = xmax - (x_range * 0.15)
+        left_side_x_pos_free_throw_line_edge = xmin + (x_range * 0.225)
+        right_side_x_pos_free_throw_line_edge = xmax - (x_range * 0.225)
+        left_side_4m_line_x_pos = xmin + (x_range * 0.1)
+        lower_y_pos_4m_line = ymin + (y_range * 0.49625)
+        upper_y_pos_4m_line = ymin + (y_range * 0.50375)
+        left_side_7m_line_x_pos = xmin + (x_range * 0.175)
+        right_side_7m_line_x_pos = xmax - (x_range * 0.175)
+        lower_y_pos_7m_line = ymin + (y_range * 0.475)
+        upper_y_pos_7m_line = ymin + (y_range * 0.525)
+        x_margin = x_range / 40
+        y_margin = y_range / 20
```

Comment:
I think its very concise to bundle all magic numbers and pitch markings here (given the nature of the task). However, a quick note would be helpful to intepret these (in form of 2-3 lines of inline comments). Are they proportional scalings of the objects locations w.r.t to the min, max, and range values?